omero/developers/testing.txt

```diff
@@ -401,93 +495,86 @@ There are a small number of custom markers defined:
 
-        ./setup.py test --markers
+New clients can be instantiated by individual tests using the
+``ITest.new_client()`` or ``ITest.new_client_and_user()`` methods::
 
-.. option:: -s
+      def testNewClient():
+          new_client = self.new_user_and_client()
+          ec = new_client.getSession().getAdminService().getEventContext()
+          assert ec, "No EventContext!"
 
-    This option allows the standard output to be shown on the console::
+Images can be imported using the ``ITest.importSingleImage()`` or
+``ITest.importSingleImageWithCompanion()`` or ``ITest.importMIF()`` methods::
 
-        ./setup.py test -t test/integration/test_admin.py -s
+      def testNewClient():
+          images = self.importMIF(2)
+          assert len(images) == 2
 
-.. option:: -h, --help
+Writing OMERO.web tests
+"""""""""""""""""""""""
 
-    This option displays the full list of available options::
+For OMERO.web integration tests, the :source:`OMERO.web test library <components/tools/OmeroWeb/test/integration/weblibrary.py>`
+defines an abstract ``IWebTest`` class that inherits from ``ITest`` and
+also implements Django clients at the class setup using the
+:djangodoc:`Django testing tools <topics/testing/tools>`.
 
-        ./setup.py test -h
+On top of the elements created by ``ITest.setup_class()``, the ``IWebTest``
+class creates:
 
-To make use of the more advanced options available in `pytest` that are not
-accessible using :program:`setup.py test`, the :program:`py.test` script can
-be used directly. To use this :envvar:`PYTHONPATH` must contain the path to
-the OMERO Python libraries, see |BlitzGateway| as well as the  path to the
-:sourcedir:`OMERO Python test library <components/tests/python>`.
-Alternatively, the `pytest` plugin :pypi:`pytest-pythonpath` can be used to
-add paths to :envvar:`PYTHONPATH` specifically for `pytest`.
-
-.. program:: py.test
+- ``self.django_root_client`` is a Django test client for the root user
+- ``self.django_client`` is a client for the new user created at the class
+  setup.
 
-.. option:: --repeat <number>
-
-    This option allows to repeat tests for *number* occurences::
+::
 
-        py.test --repeat 20 test/unit/fstest
+  from weblibrary import IWebTest
 
-.. option:: -h, --help
+  class TestExample(IWebTest):
+      def testSimple():
+          self.django_client.post('/login/', {'username': 'john'})
 
-    This option displays the full list of options::
+New Django test clients can be instantiated by individual tests using the
+``IWebTest.new_django_client()`` method::
 
-        py.test --help
+      def testNewDjangoClient():
+          new_user = self.new_user()
+          omeName = new_user.omeName.val
+          new_django_client = self.new_django_client(omeName, omeName)
 
-and `<http://pytest.org/latest/usage.html>`_ for more help in running tests.
+.. seealso::
+  :source:`test_simple.py <components/tools/OmeroWeb/test/integration/test_simple.py>`
```

Comment:
I think this line needs indenting otherwise the link is rendered right up against the left-hand side of the box. Compare with https://github.com/openmicroscopy/ome-documentation/pull/1089/files#diff-376c2e21de4aa95f3531870a1890bad7R240